packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts

```diff
@@ -189,4 +197,91 @@ describeCompat("TestSignals", "FullCompat", (getTestObjectProvider) => {
 			"client 2 did not receive signal on data store runtime",
 		);
 	});
+
+	it.skip("Validate data store runtime targeted signals", async () => {
+		let user1SignalReceivedCount = 0;
+		let user2SignalReceivedCount = 0;
+		let user3SignalReceivedCount = 0;
+
+		dataObject1.runtime.on("signal", (message: IInboundSignalMessage, local: boolean) => {
+			if (message.type === "TestSignal") {
+				user1SignalReceivedCount += 1;
+			}
+		});
+
+		dataObject2.runtime.on("signal", (message: IInboundSignalMessage, local: boolean) => {
+			if (message.type === "TestSignal") {
+				user2SignalReceivedCount += 1;
+			}
+		});
+
+		dataObject3.runtime.on("signal", (message: IInboundSignalMessage, local: boolean) => {
+			if (message.type === "TestSignal") {
+				user3SignalReceivedCount += 1;
+			}
+		});
+
+		dataObject1.runtime.submitSignal("TestSignal", true, dataObject2.runtime.clientId);
+		await waitForSignal(dataObject2.runtime);
+		assert.equal(user1SignalReceivedCount, 0, "client 1 should not receive signal");
+		assert.equal(user2SignalReceivedCount, 1, "client 2 did not receive signal");
+		assert.equal(user3SignalReceivedCount, 0, "client 3 should not receive signal");
+
+		dataObject1.runtime.submitSignal("TestSignal", true, dataObject3.runtime.clientId);
+		await waitForSignal(dataObject3.runtime);
+		assert.equal(user1SignalReceivedCount, 0, "client 1 should not receive signal");
+		assert.equal(user2SignalReceivedCount, 1, "client 2 should not receive signal");
+		assert.equal(user3SignalReceivedCount, 1, "client 3 did not receive signal");
+
+		dataObject2.runtime.submitSignal("TestSignal", true, dataObject1.runtime.clientId);
+		await waitForSignal(dataObject1.runtime);
+		assert.equal(user1SignalReceivedCount, 1, "client 1 did not receive signal");
+		assert.equal(user2SignalReceivedCount, 1, "client 2 should not receive signal");
+		assert.equal(user3SignalReceivedCount, 1, "client 3 should not receive signal");
+	});
+
+	it.skip("Validate ContainerRuntime targeted signals", async () => {
+		let user1SignalReceivedCount = 0;
+		let user2SignalReceivedCount = 0;
+		let user3SignalReceivedCount = 0;
+		const user1ContainerRuntime = dataObject1.context.containerRuntime as ContainerRuntime;
```

Comment:
Why do you need `as ContainerRuntime`? The tests above (see line 125) don't need that.
(If you ever see or use `as`, ask why it is needed. We really want to avoid them.)

Reply:
Without `as ContainerRuntime` the type defaults to ContainerRuntimeBase, which does not have the targetClientId? parameter on submitSignal signature.  Will leave a comment for now and look into other ways to avoid `as`.

Reply:
Looking at #17729, `IContainerRuntimeBase` was not amended. But `IFluidDataStoreContext` was.
We definitely should not be adding testing to internals like `ContainerRuntime` but should use exposed interfaces (often prefixed with `I` in the past). Maybe `IContainerRuntimeBase` should have been updated; maybe it was intentionally not. I am creating task to check into that - follow up with those involved in the PR to confirm. If intentionally omitted then, don't test against container for targeted signals (could test against context). If interface should be amended, do that first.